Sources/Utility/Process.swift

```diff
@@ -77,16 +78,34 @@ public final class Process: ObjectIdentifierProtocol {
 
     /// The process id of the spawned process, available after the process is launched.
     public private(set) var processID = ProcessID()
-
+    
     /// The result of the process execution. Available after process is terminated.
-    public private(set) var result: ProcessResult?
-
+    public var result: ProcessResult? {
+        get {
+            return self.serialQueue.sync {
+                self._result
+            }
+        }
+    }
+    
     /// If the subprocess has launched.
-    public private(set) var launched = false
+    public var launched: Bool {
+        get {
+            return self.serialQueue.sync {
```

Comment:
I think we don't need `launched` to be protected by a serial queue because the only place it is mutated is in `launch()` and calling that more than once is a programming error.